File: server/src/main/java/com/example/server/request/model/ItemRequest.java
package com.example.server.request.model;

import com.example.server.user.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@AllArgsConstructor
@Data
@Entity
@NoArgsConstructor
@Table(name = "requests")
public class ItemRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String description;

    @ManyToOne
    @JoinColumn(name = "requester_id")
    private User requester;
    private LocalDateTime created;
}

File: application/src/main/kotlin/com/example/server/entity/ItemRequest.kt
package com.example.server.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.SequenceGenerator
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Entity
@SequenceGenerator(name = "request_id_seq", allocationSize = 1)
@Table(name = "request")
class ItemRequest(

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "request_id_seq")
    var id: Long?,

    @Column(name = "description")
    var description: String,

    @JoinColumn(name = "requester_id")
    @ManyToOne
    var requester: User,

    @Column(name = "created")
    @CreationTimestamp
    var created: LocalDateTime?,

    @OneToMany(mappedBy = "request")
    var items: MutableList<Item>
)
